File: patterns/C++/pattern16.cpp
/*
Generate the following pattern
*                 *
* *             * *
* * *         * * *
* * * *     * * * *
* * * * * * * * * *
* * * *     * * * *
* * *         * * *
* *             * *
*                 *
*/

#include<bits/stdc++.h>
using namespace std;

void pattern16(int n){
    for(int i=0; i<n; i++){
        for(int j=0; j<=i; j++){
            cout << "*" <<" ";
        }

        for(int j=0; j<4*(n-i-1); j++){
            cout << " ";
        }

        for(int j=0; j<=i; j++){
            cout << "*" << " ";
        }

        cout << endl;
    }

    for(int i=n; i<2*n-1; i++){
        for(int j=0; j<(2*n-i-1); j++){
            cout << "*" <<" ";
        }

        for(int j=0; j<4*(i-n+1); j++){
            cout << " ";
        }

        for(int j=0; j<(2*n-i-1); j++){
            cout << "*" << " ";
        }

        cout << endl;
    }

}


int main(){
    int n;
    cout << "Enter the Number of rows: ";
    cin >> n;

    cout << "\nHere is the generated pattern\n";
    pattern16(n);
}

File: patterns/Python/pattern16.py
"""
Generate the following pattern
*                 *
* *             * *
* * *         * * *
* * * *     * * * *
* * * * * * * * * *
* * * *     * * * *
* * *         * * *
* *             * *
*                 *
"""

def pattern16(n):
    for i in range(1, n+1):
        for j in range(i):
            print("*", end=" ")
            
        for j in range(4*(n-i)):
            print(" ", end="")
            
        for j in range(i):
            print("*", end=" ")
                   
        print("")
        
    for i in range(n+1, 2*n):
        for j in range(2*n-i):
            print("*", end=" ")
            
        for j in range(4*(i-n)):
            print(" ", end="")
            
        for j in range(2*n-i):
            print("*", end=" ")
                   
        print("")
   
#Driver Code 
def main():
    n = int(input("Enter the number of rows: "))
    print("Here is the generated pattern\n")
    pattern16(n)
    
if __name__ == "__main__":
    main()
           